tests/data/dataset_readers/wikitables/wikitables_dataset_reader_test.py

```diff
@@ -11,17 +12,244 @@ def test_reader_reads(self):
+        # we're expecting are present, and there are no extras.
+        assert sorted(actions) == [
+                # Placeholder types
+                "<#1,#1> -> !=",
+                "<#1,#1> -> fb:type.object.type",
+                "<#1,#1> -> var",
+                "<#1,<#1,#1>> -> and",
+                "<#1,<#1,#1>> -> or",
+                "<#1,d> -> count",
+                "<<#1,#2>,<#2,#1>> -> reverse",
+
+                # These complex types are largely just here to build up a few specific functions:
+                # argmin and argmax.
+                "<<d,d>,d> -> [<d,<<d,d>,d>>, d]",
+                "<<d,e>,e> -> [<e,<<d,e>,e>>, e]",
+                "<<d,p>,p> -> [<p,<<d,p>,p>>, p]",
+                "<<d,r>,r> -> [<r,<<d,r>,r>>, r]",
+                "<d,<<d,d>,d>> -> [<d,<d,<<d,d>,d>>>, d]",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmax",
+                "<d,<d,<#1,<<d,#1>,#1>>>> -> argmin",
+                "<d,<d,<<d,d>,d>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<d,d>> -> -",
+                "<d,<e,<<d,e>,e>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<p,<<d,p>,p>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+                "<d,<r,<<d,r>,r>>> -> [<d,<d,<#1,<<d,#1>,#1>>>>, d]",
+
+                # Operations that manipulate numbers.
+                # TODO(mattg): question for Pradeep: why aren't these two-argument functions?
+                "<d,d> -> <",
+                "<d,d> -> <=",
+                "<d,d> -> >",
+                "<d,d> -> >=",
+                "<d,d> -> ['lambda x', d]",
+                "<d,d> -> [<#1,<#1,#1>>, d]",
+                "<d,d> -> [<<#1,#2>,<#2,#1>>, <d,d>]",
+                "<d,d> -> [<d,<d,d>>, d]",
+                # These are single-argument functions because the "d" type here actually represents
+                # a set, and we're performing some aggregation over the set and returning a single
+                # number (actually a set with a single number in it).
+                "<d,d> -> avg",
+                "<d,d> -> max",
+                "<d,d> -> min",
+                "<d,d> -> sum",
+                "<d,e> -> ['lambda x', e]",
+                "<d,e> -> [<<#1,#2>,<#2,#1>>, <e,d>]",
+                # TODO(mattg): isn't this backwards?  Aren't these functions that take cells and
+                # return numbers?
+                "<d,e> -> fb:cell.cell.date",
+                "<d,e> -> fb:cell.cell.num2",
+                "<d,e> -> fb:cell.cell.number",
+                "<d,p> -> ['lambda x', p]",
+                "<d,p> -> [<<#1,#2>,<#2,#1>>, <p,d>]",
+                "<d,r> -> ['lambda x', r]",
+                "<d,r> -> [<<#1,#2>,<#2,#1>>, <r,d>]",
+                "<d,r> -> fb:row.row.index",
+
+                # Now we get to the CELL_TYPE, which is represented by "e".
+                "<e,<<d,e>,e>> -> [<d,<e,<<d,e>,e>>>, d]",
+                # TODO(mattg): why is this so complicated?
```

Comment:
`date` takes three arguments: year, month and day. For example, `(date 2018 01 06)`.

Reply:
Why do the year, month and day have type `e` instead of `d`?